nessie/src/test/java/org/apache/iceberg/nessie/TestNamespace.java

```diff
@@ -132,6 +133,15 @@ public void testSettingProperties() {
         .hasMessage("Namespace does not exist: unknown");
   }
 
+  @Test
+  public void testEmptyNamespace() {
+    Assertions.assertThatThrownBy(
+            () -> catalog.createNamespace(Namespace.empty(), Collections.emptyMap()))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Creating empty namespaces is not supported");
+    Assertions.assertThat(catalog.namespaceExists(Namespace.empty())).isFalse();
```

Comment:
this check used to fail with the callstack mentioned in the PR description.

Reply:
I would also add a check that uses `loadNamespaceMetadata()` directly with an empty namespace